src/pubsub/mod_pubsub_db_mnesia.erl

```diff
@@ -259,20 +259,36 @@ remove_all_items(Nidx) ->
 
 -spec add_item(Nidx :: mod_pubsub:nodeIdx(),
                JID :: jid:jid(),
-               ItemId :: mod_pubsub:itemId()) ->
+               ItemId :: mod_pubsub:pubsubItem()) ->
     ok.
-add_item(Nidx, JID, ItemId) ->
+add_item(Nidx, JID, #pubsub_item{itemid = {ItemId, _}} = Item) ->
+    set_item(Item),
     {ok, State} = get_state(Nidx, jid:to_bare(JID), write),
     NewItems = [ItemId | State#pubsub_state.items],
     mnesia:write(State#pubsub_state{ items = NewItems }).
 
 %% ------------------------ Direct #pubsub_item access ------------------------
 
--spec get_items(Nidx :: mod_pubsub:nodeIdx()) ->
+-spec get_items(Nidx :: mod_pubsub:nodeIdx(), gen_pubsub_node:get_item_options()) ->
     {ok, {[mod_pubsub:pubsubItem()], none}}.
-get_items(Nidx) ->
+get_items(Nidx, Opts) ->
     Items = mnesia:match_object(#pubsub_item{itemid = {'_', Nidx}, _ = '_'}),
-    {ok, {lists:reverse(lists:keysort(#pubsub_item.modification, Items)), none}}.
+    Sorted = lists:reverse(lists:keysort(#pubsub_item.modification, Items)),
```

Comment:
How long is usually `Items` list?
Would it make sense to replace `lists:reverse(lists:keysort(...))` with proper `lists:sort/2`??

Reply:
> How long is usually Items list?
That depends on the use case, how many items one put to a node

> Would it make sense to replace lists:reverse(lists:keysort(...)) with proper lists:sort/2??
Probably it would. This is the mnesia backend we are going to replace with more efficient RDBMS backend so I think making the optimization now is less important.